server/src/index.ts

```diff
@@ -710,6 +711,27 @@ app.get(
   },
 );
 
+app.get(
+  "/api/getAllAssignmentStudentData/:userId",
+  async (req: Request, res: Response, next: NextFunction) => {
+    const userId = Number(req.params.userId);
+    //const loggedInUserId = Number(req.cookies.userId);
```

Comment:
We only want a logged in user to be able to view their own assignment data. For that reason, I recommend removing the `userId` parameter and just pass `loggedInUserId` into the function. Once we have a notion of folders or courses, then we'll pass in the id of that folder/course and allow access only if `loggedInUserId` is the owner of that folder/course.

Reply:
I pushed a new version with this change. For now I left in `userId` as a parameter that just doesn't get used because some of the other functions seemed to be passing it and I wasn't sure why/if it might come in handy in the future, but I can take it out entirely if you'd like.

Reply:
Alright, just removed it!